pyoptsparse/pySNOPT/source/f2py/snopt.pyf

```diff
@@ -396,50 +396,50 @@ python module snopt ! in
+            character*(*) intent(in) :: start
+            integer, optional, intent(in), check(len(pi)>=m),depend(pi) :: m=len(pi)
+            integer, optional, intent(in), check((len(locj)-1)>=n),depend(locj) :: n=(len(locj)-1)
+            integer, optional, intent(in), check(len(jcol)>=nej),depend(jcol) :: nej=len(jcol)
+            integer, optional,check(shape(names,0)==nnames),depend(names) :: nnames=shape(names,0)
+            integer intent(in) :: nncon
+            integer intent(in) :: nnobju
+            integer intent(in) :: nnjac
+            integer intent(in) :: iobju
+            double precision intent(in) :: objuadd
+            character*8 intent(in) :: prob
             external userfg,snlog,snlog2,sqlog,snstop
-            double precision dimension(ne) :: jcol
-            integer intent(inout), dimension(ne),depend(ne) :: indj
-            integer intent(inout), dimension(n + 1) :: locj
-            double precision intent(inout), dimension(n+m),depend(m,n) :: bl
-            double precision intent(inout), dimension(n+m),depend(m,n) :: bu
-            character*8 intent(in), dimension(nname) :: names
-            integer optional,check(len(names)==nname),depend(names) :: nname=len(names)
-            integer intent(inout), dimension(n+m),depend(m,n) :: hs
-            double precision intent(inout), dimension(n+m),depend(m,n) :: x
-            double precision intent(inout), dimension(m) :: pi
-            double precision intent(inout), dimension(n+m),depend(m,n) :: rc 
-            integer intent(inout) :: inform
-            integer intent(inout) :: mincw
-            integer intent(inout) :: miniw
-            integer intent(inout) :: minrw
-            integer :: ns
-            integer :: ninf
-            double precision :: sinf
+            double precision dimension(nej),intent(in) :: jcol
+            integer dimension(nej),intent(in),depend(nej) :: indj
+            integer dimension(n + 1),intent(in) :: locj
+            double precision dimension(n+m),intent(in),depend(m,n) :: bl
+            double precision dimension(n+m),intent(in),depend(m,n) :: bu
+            character dimension(nnames,8),intent(in) :: names
+            integer dimension(n+m),intent(in,out),depend(m,n) :: hs
+            double precision dimension(n+m),intent(in,out),depend(m,n) :: x
+            double precision dimension(m),intent(in,out) :: pi
+            double precision dimension(n+m),intent(out),depend(m,n) :: rc
+            integer intent(out) :: info
+            integer intent(out) :: mincw
+            integer intent(out) :: miniw
+            integer intent(out) :: minrw
+            integer intent(out) :: ns
+            integer intent(out) :: ninf
+            double precision intent(out) :: sinf
             double precision intent(out) :: obj
-            character*8 intent(inplace), dimension(lencu) :: cu
-            integer optional,check(len(cu)==lencu),depend(cu) :: lencu=len(cu)
```

Comment:
Are we changing the shape of this array with this PR?

Reply:
Are you talking about the array `cw`? No, it is still `lencu`, but the size of each entry was previously specified via `character*8` and now via the second argument to `dimension(lencu, 8)`. I think this is the preferred, more modern approach.